deployment/docker/Makefile: switch base Docker image from Alpine to distroless (#1494)
VictoriaMetrics/VictoriaLogs: docs/victorialogs/CHANGELOG.md

## tip

> **Update note 1:** the base Docker image has been changed from Alpine to `distroless`. If you relied on Alpine-specific tools or shell access inside the container, it is recommended to use Alpine-based image directly instead. For debugging in Kubernetes it is recommended to use `kubectl debug`.

Reviewer: ```suggestion
**Update note 1:** the base Docker image has been changed from [Alpine](https://www.alpinelinux.org/) to [distroless](https://github.com/googlecontainertools/distroless). For debugging VictoriaLogs containers in Kubernetes it is recommended to use [`kubectl debug`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_debug/).
```

> **Update note 1:** the base Docker image has been changed from [Alpine](https://www.alpinelinux.org/) to [distroless](https://github.com/googlecontainertools/distroless). For debugging VictoriaLogs containers in Kubernetes it is recommended to use [`kubectl debug`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_debug/).

Reviewer: ```suggestion
**Update note 1:** the base Docker image has been changed from [Alpine](https://www.alpinelinux.org/) to [distroless](https://github.com/googlecontainertools/distroless) in order to reduce the attack surface. For debugging VictoriaLogs containers in Kubernetes it is recommended to use [`kubectl debug`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_debug/).
```

> **Update note 2:** VictoriaLogs no longer provides a Docker image for the `linux/386` platform. Users can build the binary for this platform themselves using the `make <appname>-linux-386-prod` command.

Reviewer: ```suggestion
**Update note 2:** VictoriaLogs no longer provides a Docker image for the `linux/386` platform because the `distroless` base image [doesn't support this platform](https://github.com/GoogleContainerTools/distroless/issues/881). Executable files for `linux/386` platform are still published at [the VictoriaLogs releases page](https://github.com/VictoriaMetrics/VictoriaLogs/releases).
```

* FEATURE: [dashboards/kubernetes-explorer](https://github.com/VictoriaMetrics/VictoriaLogs/blob/master/dashboards/victorialogs-kubernetes-explorer.json): add new dashboard for exploring Kubernetes logs via [VictoriaLogs datasource](https://docs.victoriametrics.com/victorialogs/integrations/grafana/) in Grafana. Thanks to @sias32 for [the contribution](https://github.com/VictoriaMetrics/VictoriaLogs/pull/1254)!
* FEATURE: [File Collector](https://docs.victoriametrics.com/victorialogs/vlagent/#collect-logs-from-files): enhance glob pattern support with additional syntax: double-star `/**/` for matching nested directories (e.g. `/var/log/**/*.log`), alternatives `{a,b}` for matching multiple specific names (e.g. `{access,error}.log`), character classes `[a-z]` and `?` wildcard for single-character matching. See [#1393](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1393) and [these docs](https://docs.victoriametrics.com/victorialogs/vlagent/#glob-pattern-requirements) for the full pattern syntax reference.

> * FEATURE: switch base Docker image from Alpine to `distroless` for VictoriaLogs, `vlagent` and `vlogscli`. This reduces the image size and attack surface. See [#1228](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1228).

Reviewer: ```suggestion
* FEATURE: switch base Docker image from [Alpine](https://www.alpinelinux.org/) to [distroless](https://github.com/GoogleContainerTools/distroless/) for VictoriaLogs, `vlagent` and `vlogscli`. This reduces the image size and attack surface. See [#1228](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1228).
```
